src/color.rs

```diff
@@ -0,0 +1,522 @@
+    swizzle!(b, g, b, g);
+    swizzle!(b, g, b, b);
+    swizzle!(b, g, b, a);
+    swizzle!(b, g, a, r);
+    swizzle!(b, g, a, g);
+    swizzle!(b, g, a, b);
+    swizzle!(b, g, a, a);
+    swizzle!(b, b, r, r);
+    swizzle!(b, b, r, g);
+    swizzle!(b, b, r, b);
+    swizzle!(b, b, r, a);
+    swizzle!(b, b, g, r);
+    swizzle!(b, b, g, g);
+    swizzle!(b, b, g, b);
+    swizzle!(b, b, g, a);
+    swizzle!(b, b, b, r);
+    swizzle!(b, b, b, g);
+    swizzle!(b, b, b, b);
+    swizzle!(b, b, b, a);
+    swizzle!(b, b, a, r);
+    swizzle!(b, b, a, g);
+    swizzle!(b, b, a, b);
+    swizzle!(b, b, a, a);
+    swizzle!(b, a, r, r);
+    swizzle!(b, a, r, g);
+    swizzle!(b, a, r, b);
+    swizzle!(b, a, r, a);
+    swizzle!(b, a, g, r);
+    swizzle!(b, a, g, g);
+    swizzle!(b, a, g, b);
+    swizzle!(b, a, g, a);
+    swizzle!(b, a, b, r);
+    swizzle!(b, a, b, g);
+    swizzle!(b, a, b, b);
+    swizzle!(b, a, b, a);
+    swizzle!(b, a, a, r);
+    swizzle!(b, a, a, g);
+    swizzle!(b, a, a, b);
+    swizzle!(b, a, a, a);
+    swizzle!(a, r, r, r);
+    swizzle!(a, r, r, g);
+    swizzle!(a, r, r, b);
+    swizzle!(a, r, r, a);
+    swizzle!(a, r, g, r);
+    swizzle!(a, r, g, g);
+    swizzle!(a, r, g, b);
+    swizzle!(a, r, g, a);
+    swizzle!(a, r, b, r);
+    swizzle!(a, r, b, g);
+    swizzle!(a, r, b, b);
+    swizzle!(a, r, b, a);
+    swizzle!(a, r, a, r);
+    swizzle!(a, r, a, g);
+    swizzle!(a, r, a, b);
+    swizzle!(a, r, a, a);
+    swizzle!(a, g, r, r);
+    swizzle!(a, g, r, g);
+    swizzle!(a, g, r, b);
+    swizzle!(a, g, r, a);
+    swizzle!(a, g, g, r);
+    swizzle!(a, g, g, g);
+    swizzle!(a, g, g, b);
+    swizzle!(a, g, g, a);
+    swizzle!(a, g, b, r);
+    swizzle!(a, g, b, g);
+    swizzle!(a, g, b, b);
+    swizzle!(a, g, b, a);
+    swizzle!(a, g, a, r);
+    swizzle!(a, g, a, g);
+    swizzle!(a, g, a, b);
+    swizzle!(a, g, a, a);
+    swizzle!(a, b, r, r);
+    swizzle!(a, b, r, g);
+    swizzle!(a, b, r, b);
+    swizzle!(a, b, r, a);
+    swizzle!(a, b, g, r);
+    swizzle!(a, b, g, g);
+    swizzle!(a, b, g, b);
+    swizzle!(a, b, g, a);
+    swizzle!(a, b, b, r);
+    swizzle!(a, b, b, g);
+    swizzle!(a, b, b, b);
+    swizzle!(a, b, b, a);
+    swizzle!(a, b, a, r);
+    swizzle!(a, b, a, g);
+    swizzle!(a, b, a, b);
+    swizzle!(a, b, a, a);
+    swizzle!(a, a, r, r);
+    swizzle!(a, a, r, g);
+    swizzle!(a, a, r, b);
+    swizzle!(a, a, r, a);
+    swizzle!(a, a, g, r);
+    swizzle!(a, a, g, g);
+    swizzle!(a, a, g, b);
+    swizzle!(a, a, g, a);
+    swizzle!(a, a, b, r);
+    swizzle!(a, a, b, g);
+    swizzle!(a, a, b, b);
+    swizzle!(a, a, b, a);
+    swizzle!(a, a, a, r);
+    swizzle!(a, a, a, g);
+    swizzle!(a, a, a, b);
+    swizzle!(a, a, a, a);
+}
+
+impl_op_ex!(+= |a: &mut Color, b: &Color| { a.r += b.r; a.g += b.g; a.b += b.b; a.a += b.a; });
+impl_op_ex!(-= |a: &mut Color, b: &Color| { a.r -= b.r; a.g -= b.g; a.b -= b.b; a.a -= b.a; });
```

Comment:
What happens when one of the components gets <0 ? The field docs specify the components to be [0.0-1.0]

Reply:
Maybe we should allow all ranges, to keep it simple.

Reply:
This would probably be the best solution